code/go/0chain.net/blobbercore/datastore/store.go

```diff
@@ -31,7 +31,8 @@ type Store interface {
 	CreateTransaction(ctx context.Context) context.Context
 	// GetTransaction get transaction from context
 	GetTransaction(ctx context.Context) *EnhancedDB
-	WithTransaction(ctx context.Context, tx *gorm.DB) context.Context
+	WithNewTransaction(f func(ctx context.Context) error) error
+	WithTransaction(ctx context.Context, f func(ctx context.Context) error) error
```

Comment:
What is the difference between these two methods? And I do not see any usage of the `WithTransaction` method. Did I not check it properly? If it indeed do not have any usage, is it really required then?

Reply:
they are different actually
`WithNewTransaction(f func(ctx context.Context) error) error` will force create new transaction and new context
`WithTransaction(ctx context.Context, f func(ctx context.Context) error) error` will use existent transaction or will create the new one in given context